code/modules/reagents/reagent_containers/food/snacks/egg.dm

```diff
@@ -4,12 +4,21 @@
 	name = "egg"
 	desc = "An egg!"
 	icon_state = "egg"
+	var/amount_grown = 0
 
 /obj/item/weapon/reagent_containers/food/snacks/egg/New()
 	..()
 	reagents.add_reagent("nutriment", 4)
 	src.bitesize = 3
 
+/obj/item/weapon/reagent_containers/food/snacks/egg/process()
+	if(is_in_valid_nest(src)) //_macros.dm
```

Comment:
I'm pretty sure that's to make it so it doesn't hatch inside of a bag/pocket, though that could be handled by changing where the new chicken is spawned.